[Feature] Render error card in same theme as requested card #3259

notpritam opened the pull request:
Hello , i have add the feature

here is the before and after

```js
return res.send(renderError(err.message, err.secondaryMessage));
```

i have added it by introducing a new params in renderError function :- 

```js
    return res.send(
      renderError(err.message, err.secondaryMessage, {
        title_color,
        text_color,
        bg_color,
        border_color,
        theme,
      }),
    );
```


[Previous](https://github-readme-stats.vercel.app/api?username=notprita&hide=contribs,prs&theme=transparent) 
[Example 1](https://github-readme-stats-azure-zeta.vercel.app/api?username=notprita&hide=contribs,prs&theme=transparent)
[Previous 2](https://github-readme-stats.vercel.app/api?username=notprita&hide=contribs,prs&theme=merko) 
[Example 2](https://github-readme-stats-azure-zeta.vercel.app/api?username=notprita&hide=contribs,prs&theme=merko)

qwerty541 reviewed (approved):
Hey, @notpritam! Thanks for your pull request and for your patience! Generally your code looks well, but it would be nice if this feature can be covered with test. Also i'm okay with adding test in future pull requests since i see that this feature works on test deployment that you provided. I give my approval to this pull request and mark it with https://github.com/anuraghazra/github-readme-stats/labels/hacktoberfest-accepted label. We will merge it when any other maintainer also gives his approval.

qwerty541 commented:
> > Hey, @notpritam! Thanks for your pull request and for your patience! Generally your code looks well, but it would be nice if this feature can be covered with test.
> 
> Thanks for your response , i dont have any experience creating test so i didnt created that. if you want to me create the test for this also, i will love to do that but i will need a little longer time, cause this will be my first time creating any tests.

As I said, this is desirable, not required. I hope that adding this test case into https://github.com/anuraghazra/github-readme-stats/blob/master/tests/api.test.js file will be enough, may be some changes will be needed, or i can add personally in future PRs.

```js
it("should render error card in provided theme", async () => {
    const { req, res } = faker({ theme: "merko" }, error);

    await api(req, res);

    expect(res.setHeader).toBeCalledWith("Content-Type", "image/svg+xml");
    expect(res.send).toBeCalledWith(
      renderError(
        error.errors[0].message,
        "Make sure the provided username is not an organization",
        { theme: "merko" },
      ),
    );
  });
```

I also noticed that there is some conflicts with master branch, can you please resolve it?

notpritam commented:
> > > Hey, @notpritam! Thanks for your pull request and for your patience! Generally your code looks well, but it would be nice if this feature can be covered with test.
> > 
> > 
> > Thanks for your response , i dont have any experience creating test so i didnt created that. if you want to me create the test for this also, i will love to do that but i will need a little longer time, cause this will be my first time creating any tests.
> 
> As I said, this is desirable, not required, since you provided a link on test vercel deployment. I hope that adding this test case into https://github.com/anuraghazra/github-readme-stats/blob/master/tests/api.test.js file will be enough, may be some changes will be needed, or i can add it personally in future PRs.
> 
> ```js
> it("should render error card in provided theme", async () => {
>     const { req, res } = faker({ theme: "merko" }, error);
> 
>     await api(req, res);
> 
>     expect(res.setHeader).toBeCalledWith("Content-Type", "image/svg+xml");
>     expect(res.send).toBeCalledWith(
>       renderError(
>         error.errors[0].message,
>         "Make sure the provided username is not an organization",
>         { theme: "merko" },
>       ),
>     );
>   });
> ```
> 
> I also noticed that there is some conflicts with master branch, can you please resolve it?

Hii, @qwerty541 , i have fixed the conflicts, and also updated the code, please review the and merge it ,

thanks...

notpritam commented:
@rickstaa and @anuraghazra  , can you please look over the code and approve it  , 
thanks a lot...

qwerty541 commented on src/common/utils.js:
```diff
@@ -335,16 +375,20 @@ const renderError = (message, secondaryMessage = "", options = {}) => {
 
   return `
-    <svg width="${ERROR_CARD_LENGTH}"  height="120" viewBox="0 0 ${ERROR_CARD_LENGTH} 120" fill="${bgColor}" xmlns="http://www.w3.org/2000/svg">
+    <svg wwidth="${ERROR_CARD_LENGTH}"  height="120" viewBox="0 0 ${ERROR_CARD_LENGTH} 120" fill="${bgColor}" xmlns="http://www.w3.org/2000/svg">
```
```suggestion
    <svg width="${ERROR_CARD_LENGTH}"  height="120" viewBox="0 0 ${ERROR_CARD_LENGTH} 120" fill="${bgColor}" xmlns="http://www.w3.org/2000/svg">
```

qwerty541 commented on src/common/utils.js:
```diff
@@ -335,16 +375,20 @@ const renderError = (message, secondaryMessage = "", options = {}) => {
     <style>
     .text { font: 600 16px 'Segoe UI', Ubuntu, Sans-Serif; fill: ${titleColor} }
-    .small { font: 600 12px 'Segoe UI', Ubuntu, Sans-Serif; fill: ${textColor} }
+    .small { font: 600 12px 'Segoe UI', Ubuntu, Sans-Serif; fill: ${textColor}}
```
```suggestion
    .small { font: 600 12px 'Segoe UI', Ubuntu, Sans-Serif; fill: ${textColor} }
```

qwerty541 reviewed (changes requested):
> > > > Hey, @notpritam! Thanks for your pull request and for your patience! Generally your code looks well, but it would be nice if this feature can be covered with test.
> > > 
> > > 
> > > Thanks for your response , i dont have any experience creating test so i didnt created that. if you want to me create the test for this also, i will love to do that but i will need a little longer time, cause this will be my first time creating any tests.
> > 
> > 
> > As I said, this is desirable, not required, since you provided a link on test vercel deployment. I hope that adding this test case into https://github.com/anuraghazra/github-readme-stats/blob/master/tests/api.test.js file will be enough, may be some changes will be needed, or i can add it personally in future PRs.
> > ```
> > it("should render error card in provided theme", async () => {
> >     const { req, res } = faker({ theme: "merko" }, error);
> > 
> >     await api(req, res);
> > 
> >     expect(res.setHeader).toBeCalledWith("Content-Type", "image/svg+xml");
> >     expect(res.send).toBeCalledWith(
> >       renderError(
> >         error.errors[0].message,
> >         "Make sure the provided username is not an organization",
> >         { theme: "merko" },
> >       ),
> >     );
> >   });
> > ```
> > 
> > 
> > I also noticed that there is some conflicts with master branch, can you please resolve it?
> 
> Hii, @qwerty541 , i have fixed the conflicts, and also updated the code, please review the and merge it ,
> 
> thanks...

@notpritam I have found two more problems after resolving conflicts, please fix it.

qwerty541 commented:
> @rickstaa and @anuraghazra , can you please look over the code and approve it , thanks a lot...

I already requested their review https://github.com/anuraghazra/github-readme-stats/pull/3298#event-10619878228

rickstaa commented:
> > @rickstaa and @anuraghazra , can you please look over the code and approve it , thanks a lot...
> 
> I already requested their review [#3298 (comment)](https://github.com/anuraghazra/github-readme-stats/pull/3298#event-10619878228)

Tested and approved! @notpritam, thanks a lot for this outstanding contribution ❤️‍🔥!

@qwerty541 we can merge this when tests are added 👍🏻.

notpritam commented:
> > > @rickstaa and @anuraghazra , can you please look over the code and approve it , thanks a lot...
> > 
> > 
> > I already requested their review [#3298 (comment)](https://github.com/anuraghazra/github-readme-stats/pull/3298#event-10619878228)
> 
> Tested and approved! @notpritam, thanks a lot for this outstanding contribution ❤️‍🔥!
> 
> @qwerty541 we can merge this when tests are added 👍🏻.

thanks a lot, means a lot for me coming from you..

qwerty541 commented:
> > > @rickstaa and @anuraghazra , can you please look over the code and approve it , thanks a lot...
> > 
> > 
> > I already requested their review [#3298 (comment)](https://github.com/anuraghazra/github-readme-stats/pull/3298#event-10619878228)
> 
> Tested and approved! @notpritam, thanks a lot for this outstanding contribution ❤️‍🔥!
> 
> @qwerty541 we can merge this when tests are added 👍🏻.

@rickstaa Since @notpritam tells that he has no experience in tests addition, may be we can merge this and i will add tests in next pull request. Anyway this feature was tested on Vercel test deployment.

rickstaa commented:
> > > > @rickstaa and @anuraghazra , can you please look over the code and approve it , thanks a lot...
> > > 
> > > 
> > > I already requested their review [#3298 (comment)](https://github.com/anuraghazra/github-readme-stats/pull/3298#event-10619878228)
> > 
> > 
> > Tested and approved! @notpritam, thanks a lot for this outstanding contribution ❤️‍🔥!
> > @qwerty541 we can merge this when tests are added 👍🏻.
> 
> @rickstaa Since @notpritam tells that he has no experience in tests addition, may be we can merge this and i will add tests in next pull request. Anyway this feature was tested on Vercel test deployment.

Sounds okay to me 👍🏻. Thanks!

qwerty541 commented:
> > > > > @rickstaa and @anuraghazra , can you please look over the code and approve it , thanks a lot...
> > > > 
> > > > 
> > > > I already requested their review [#3298 (comment)](https://github.com/anuraghazra/github-readme-stats/pull/3298#event-10619878228)
> > > 
> > > 
> > > Tested and approved! @notpritam, thanks a lot for this outstanding contribution ❤️‍🔥!
> > > @qwerty541 we can merge this when tests are added 👍🏻.
> > 
> > 
> > @rickstaa Since @notpritam tells that he has no experience in tests addition, may be we can merge this and i will add tests in next pull request. Anyway this feature was tested on Vercel test deployment.
> 
> Sounds okay to me 👍🏻. Thanks!

Thanks for approval, merged into master, i will open pull request with test addition soon.
